modules/yugabytedb/yugabytedb.go

```diff
@@ -0,0 +1,246 @@
+//			return err
+//		}
+//	}()
+//
+//	connStr, err := y.YSQLConnectionString(ctx)
+//	if err != nil {
+//		return err
+//	}
+//
+//	db, err := sql.Open("postgres", connStr)
+//	if err != nil {
+//		return err
+//	}
+//	defer db.Close()
+//
+// # Example using YCQL dialect:
+//
+//	ctx := context.Background()
+//	y, err := yugabytedb.Run(ctx, "yugabytedb/yugabyte")
+//	if err != nil {
+//		return err
+//	}
+//	defer func() {
+//		if err := testcontainers.TerminateContainer(y); err != nil {
+//			return err
+//		}
+//	}()
+//
+//	cluster := gocql.NewCluster()
+//	y.YCQLConfigureClusterConfig(ctx, cluster)
+//
+//	session, err := cluster.CreateSession()
+//	if err != nil {
+//		return err
+//	}
+//	defer session.Close()
+func Run(ctx context.Context, img string, opts ...testcontainers.ContainerCustomizer) (*YugabyteDBContainer, error) {
+	req := testcontainers.ContainerRequest{
+		Image: img,
+		Cmd:   []string{"bin/yugabyted", "start", "--background=false"},
+		WaitingFor: wait.ForAll(
+			wait.ForLog("YugabyteDB Started").WithOccurrence(1),
+			wait.ForListeningPort(ysqlPort),
+			wait.ForListeningPort(ycqlPort),
+		),
+		ExposedPorts: []string{ycqlPort, ysqlPort},
+		Env: map[string]string{
+			ycqlKeyspaceEnv:         ycqlKeyspace,
+			ycqlUserNameEnv:         ycqlUserName,
+			ycqlPasswordEnv:         ycqlPassword,
+			ysqlDatabaseNameEnv:     ysqlDatabaseName,
+			ysqlDatabaseUserEnv:     ysqlDatabaseUser,
+			ysqlDatabasePasswordEnv: ysqlDatabasePassword,
+		},
+	}
+
+	genericContainerReq := testcontainers.GenericContainerRequest{
+		ContainerRequest: req,
+		Started:          true,
+	}
+
+	for _, opt := range opts {
+		if err := opt.Customize(&genericContainerReq); err != nil {
+			return nil, fmt.Errorf("customize: %w", err)
+		}
+	}
+
+	container, err := testcontainers.GenericContainer(ctx, genericContainerReq)
+	var c *YugabyteDBContainer
+	if container != nil {
+		c = &YugabyteDBContainer{
+			Container:            container,
+			ysqlDatabaseName:     req.Env[ysqlDatabaseNameEnv],
+			ysqlDatabaseUser:     req.Env[ysqlDatabaseUserEnv],
+			ysqlDatabasePassword: req.Env[ysqlDatabasePasswordEnv],
+		}
+	}
+
+	if err != nil {
+		return c, fmt.Errorf("generic container: %w", err)
+	}
+
+	return c, nil
+}
+
+// YSQLConnectionString returns a connection string for the yugabyteDB container
+// using the configured database name, user, password, port, host and additional
+// arguments.
+// Additional arguments are appended to the connection string as query parameters
+// in the form of key=value pairs separated by "&".
+//
+// # Example:
+//
+//	connStr, err := y.YSQLConnectionString(ctx, "sslmode=disable")
+//	if err != nil {
+//		return err
+//	}
+//
+// The above code will return a connection string with the sslmode set to disable.
+func (y *YugabyteDBContainer) YSQLConnectionString(ctx context.Context, args ...string) (string, error) {
+	host, err := y.Host(ctx)
+	if err != nil {
+		return "", err
```

Comment:
suggestion: can we wrap the error so its clear to the user the source, more below
```suggestion
		return "", fmt.Errorf("host: %w", err)
```

Reply:
Attended ✅